Assignment2Q1.java

```diff
@@ -0,0 +1,97 @@
+import java.util.LinkedList;
+
+/**
+ * Created by Alexandra on 23.05.2017.
+ */
+public class Assignment2Q1 {
+
+    public static class TreeNode <T> {
+        private TreeNode<T> left;
+        private TreeNode<T> right;
+        private T value;
+
+        public TreeNode(TreeNode<T> left, TreeNode<T> right, T value) {
+            if (value == null) {
+                throw new IllegalArgumentException();
+            }
+            this.left = left;
+            this.right = right;
+            this.value = value;
+        }
+
+        public TreeNode<T> getLeft() {
+            return left;
+        }
+
+        public TreeNode<T> getRight() {
+            return right;
+        }
+
+        public T getValue() {
+            return value;
+        }
+
+        public String toString() {
+            return value.toString() + '\t';
+        }
+
+        public void printAncestors(T value) {
+            if (this.value.equals(value)) {
+                System.out.println("No ancestors."); // this node is the root node
+                return;
+            }
+            LinkedList<TreeNode<T>> ancestors = new LinkedList<>(); // stack of ancestors
+            LinkedList<Integer> checkedChildren = new LinkedList<>(); //keeps number of checked children for each ancestor
+            ancestors.add(this);
+            checkedChildren.add(0);
+            while (!ancestors.isEmpty()) {

```

Comment:
This way looks like it works, but if this was Google code I'd suggest a number of improvements:
1) If you have two very closely linked data structures, like here where you have two lists which always have to be the same length, it is more object oriented to introduce an object, and keep a single list of one type. So here you could do:
class TreePoint {
  TreeNode<T> node;
  int checkedChildren;
}
And then just have List<TreePoint> ancestors. That stops bugs happening where the two lists get out of sync, it is very useful for preventing bugs.

A design which is slightly worse than the list of objects, but better than the implementation you have is for checkedChildren to be a Map<TreeNode<T>, Integer>. I'd use a hashmap, which requires you to write equals() and hashcode() methods. Using a Map makes it clear the connection between the two data structures, rather than having an implied connection based on indexes.